### PR TITLE
Fix #42839. avoid ui freeze when file is larger than heap size limit.

### LINKED ISSUE #42839: Renderer freezes after opening 2GB or larger files
- VSCode Version: Stable/Insiders
- OS Version: macOS

Steps to Reproduce:

1. Open a 1.5GB file (with Line Model or Piece Tree)
2. It opens successfully (say the machine has enough memory)
3. Open a 2.1 GB file
4. It freezes after 1+ mins.

As we load the whole file to memory, it exceeds the memory limit of v8 process. We can expose `max_old_space_size ` flag in cli then users can increase the limit themselves.

### PR DESCRIPTION
**Pull request by rebornix**
We set the file size limit to 300MB (on 32bit) and 16GB (on 64bit) based memory optimization and analysis in https://github.com/Microsoft/vscode/issues/30180 . As described in https://github.com/Microsoft/vscode/issues/42839, v8 heap size limit is making this limit smaller on 64bit. In this PR, we check `max_old_space_size` first and if it's exceeding v8's limit, we show error message to users and ask them to run `code --max_old_space_size={LARGERSIZE}` to workaround.

@alexandrudima Can we remove the general file size limit, as we are moving to the new buffer and v8 is the limit ?

cc @bpasero

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -162,6 +162,21 @@ function touch(file) {
 	});
 }
 
+function resolveJSFlags() {
+	let jsFlags = [];
+	if (args['js-flags']) {
+		jsFlags.push(args['js-flag']);
```
Comment (bpasero) [inline src/main.js]: @rebornix typo? `js-flag` vs `js-flags`

File: src/vs/platform/environment/node/argv.ts [hunk 2]
```diff
@@ -167,6 +167,7 @@ const troubleshootingHelp: { [name: string]: string; } = {
 	'--inspect-brk-extensions': localize('inspect-brk-extensions', "Allow debugging and profiling of extensions with the extension host being paused after start. Check the developer tools for the connection uri."),
 	'--disable-gpu': localize('disableGPU', "Disable GPU hardware acceleration."),
 	'--upload-logs': localize('uploadLogs', "Uploads logs from current session to a secure endpoint."),
+	'--max_old_space_size': localize('maxOldSpaceSize', "Max size of the old space (in Mbytes).")
```
Comment (bpasero) [inline src/vs/platform/environment/node/argv.ts]: @rebornix this is a very technical description that no user will understand

**Comment by alexandrudima** [comment]
Interesting, I didn't know about this limit built into v8.

Why should this be a command line flag exposed to the user? IMHO folks would prefer large files to just work, and not have to read about the command line flag. IMHO we should simply set the value of the flag to a large number. I would suggest we also ask on the v8 mailing list why is that built-in limitation so low, and what are the consequences of raising it.

**Comment by rebornix** [comment]
From v8 mailing list:

> over time the heap size will tend to grow towards the limit.  V8 likes 
to grow the heap if it would otherwise have to scan it frequently; it 
leaves more CPU time for useful work. 

so we keep the default old_space_size_limit and change it when users open large files. The command line args will be named as `--max-memory`.

**Pull request merged**